src/Init/Notation.lean

```diff
@@ -687,6 +687,12 @@ syntax (name := checkSimp) "#check_simp " term "~>" term : command
 -/
 syntax (name := checkSimpFailure) "#check_simp " term "!~>" : command
 
+/--
+#discr_tree_key t prints the discrimination tree key for the term `t`, or if `t`
+is an identifier, for its type.
```

Comment:
I misread the code earlier, and the examples clarify the semantics when you pass a term that is not the identifier. So maybe
```suggestion
`#discr_tree_key i` prints the discrimination tree key for the theorem with name `i` (which should be an equality).

`#discr_tree_key t` prints the discrimination tree key for the type `t` (which should be the type of a simp theorem).
```

It feels a bit weird that you have to give the RHS even though it’s ignored. I would have expected a syntax like
```
#discr_tree_key List.map f (List.map g l)
```
or
```
#discr_tree_key (f : a -> b) (g : b -> c) (l : List a) : List.map f (List.map g l)
```
to get the key for an expression. Note that DiscrTrees are useful outside of simp (instance search for example).

This also indicates that maybe the other form, where you give an identifier of a simp theorem, should be its own command
```
#discr_tree_key_of_simp_lemma List.filter_map
```

Reply:
I agree. One of the main tripping points I've seen is instance synthesis. So I would certainly want something to handle that also. I started with this to be more incremental but I don't think there is much work to get this functionality. 

What other functionality do you think might be useful? Identification of where a matching succeeds or fails would be good but might be better left for the future.

Reply:
I separated them into two commands and tactics mainly because there is no `config` syntax in `Init.Notation`. 

I couldn't get the anti-quotation syntax quite right to unify the two branches for each elaboration function. Suggestions welcome.